builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,104 @@
+
+The `call()` function accepts ABI encoded call data that is then passed to the target contract. To get the encoded call data, you can use any of the [ABI encoding functions outlined in the Solidity docs](https://docs.soliditylang.org/en/latest/units-and-global-variables.html#abi-encoding-and-decoding-functions){target=_blank}, including `abi.encodeWithSelector` which will be used in the examples in this guide. 
+
+As previously mentioned, one major concern of allowing low-level calls to arbitrarily execute code on Moonbeam is that Moonbeam has precompile contracts that can be called, which can be used to get around some protections that are typically available on Ethereum. To safely use arbitrary code execution on Moonbeam, you should consider the following, which **only applies to contracts that allow arbitrary code execution**:
+
+- Moonbeam [precompiled contracts](builders/pallets-precompiles/precompiles/){target=_blank} such as the Native ERC-20 precompile, XC-20 precompiles, and XCM-related precompiles allow users to manage and transfer assets without requiring access to the EVM. Instead, these actions are done using the Substrate API. So, if your contract holds native tokens or XC-20s and allows arbitrary code execution, these precompiles can be used to drain the balance of the contract
+- Setting a value of `0` when using the `call()` function (for example, `call{value: 0}(...)`) can be overridden by calling the native asset precompile and specifying an amount to transfer in the encoded call data
+- Providing the function selector for the function that you want to allow to be executed is considered safe, assuming that the function call is safe and does not call precompiles
+- Blacklisting contracts that can execute arbitrary call data is not considered safe, as other precompiles might be added in the future. Providing whitelisted contracts that can execute the arbitrary call data is considered safe, assuming that the contracts being called are not precompiles, or that in the case they are, the contract making the call does not hold the native token or any XC-20
+
+In the following sections, you'll learn about each of these security considerations through examples.
+
+### Precompiles Can Override a Set Value {: #setting-a-value }
+
+On Ethereum, a smart contract that allows for arbitrary code execution could force the value of a call to be `0`, guaranteeing that no amount of native currency would be sent with the transaction. Whereas on Moonbeam, setting the value of an arbitrary call to `0` can be overridden by calling the native ERC-20 precompile contract and transferring the native asset through the Substrate API. Since ERC-20s and XC-20s are not native assets, setting a value to `0` doesn't provide any protection for these types of assets on Ethereum or Moonbeam.
```

Comment:
```suggestion
On Ethereum, a smart contract that allows for arbitrary code execution could force the value of a call to be a specific amount (for example, `{value: 0}`), guaranteeing that only that amount of native currency would be sent with the transaction. Whereas on Moonbeam, setting the value of an arbitrary call can be overridden by calling the native ERC-20 precompile contract and transferring the native asset through the Substrate API. Since ERC-20s and XC-20s are not native assets, setting the value attribute doesn't provide any protection for these types of assets on Ethereum or Moonbeam.
```